blueprints/controllers/ikea_e1743/ikea_e1743.yaml

```
@@ -249,24 +244,26 @@ variables:
   button_down_long: '{{ actions_mapping[integration_id]["button_down_long"] }}'
   button_down_release: '{{ actions_mapping[integration_id]["button_down_release"] }}'
   # integrations which need to store the previous press event to determine which button was released
-  #YAR - CHANGED  integrations_with_prev_event_storage: [zha, zigbee2mqtt]
-  #YAR - ADD - To Support Zigbee2MQTT_V2 with legacy=false
-  integrations_with_prev_event_storage: [zha, zigbee2mqtt, zigbee2mqtt_v2]
-  #YAR - ADD END
+  integrations_with_prev_event_storage: [zha, zigbee2mqtt]
   # build data to send within a controller event
-  controller_entity: !input controller_entity
   controller_device: !input controller_device
-  controller_id: '{% if integration_id=="zigbee2mqtt" %}{{controller_entity}}{% else %}{{controller_device}}{% endif %}'
+  controller_entity: !input controller_entity
+#**YAR** ADD - Logic to handle z2m legacy
+#**YAR** If user provided input for controller_entity assume legacy **Maybe There Is A Better Way To Do This**
+#**YAR** setting z2m_legacy to true with below code and set controller_id accordingly
+  z2m_legacy: '{{ integration_id == "zigbee2mqtt" and controller_entity not in [''] }}'
```

I get an error from this line when testing it.

Cause you escape the " jinja code" with single quotes, then you need to use double inside the jinja code or vice versa.

---

Very strange its the exact same code in the ikea_e2201 and is working for me.
Wait, I think it is because I'm not importing the blueprint but using it as is directly.
I'll check when I get home today.

So the line should be either:

z2m_legacy: "{{ integration_id == "zigbee2mqtt" and controller_entity not in [''] }}"
or
z2m_legacy: '{{ integration_id == "zigbee2mqtt" and controller_entity not in [""] }}'

---

Thanks for spotting this @Nicolai-. Agree. Not sure whether it's needed, but I'd also check for `None` and `undefined`
```suggestion
  z2m_legacy: '{{ integration_id == "zigbee2mqtt" and controller_entity not in ["", None, undefined] }}'
```

---

Done. Will test when I get back home